### PR TITLE
Added connection timeout for get_token() for DefectDojo integration

### PR DESCRIPTION
**Pull request by jeremychoi**

### CODE REVIEW

File: exports/defect_dojo.py [hunk 1]
```diff
@@ -35,18 +35,19 @@ def get_token(self):
...
-            self.token = json.load(resp)["token"]
+            logging.debug(f"resp: {resp.json()}")
+            self.token = resp.json()["token"]
 
-        self.headers["Authorization"] = f"Token {self.token}"
-        logging.debug("Defect Dojo: successfully refreshed token")
+            self.headers["Authorization"] = f"Token {self.token}"
+            logging.debug("Defect Dojo: successfully refreshed token")
+        except requests.exceptions.HTTPError as e:
+            logging.warning(f"Defect Dojo export failed: {e}")
```
[the remark this answers is not in the document]
Comment (cedricbu) [reply exports/defect_dojo.py:49]: Possibly adding `logging.error("Please verify username/password/url in 'config.defectDojo'")` ?

**Comment by jeremychoi** [comment]
@cedricbu, Thanks for the comments. I now tried updating with some better return handling.  have another look pls.

File: exports/defect_dojo.py [hunk 2]
```diff
@@ -88,18 +100,30 @@ def _private_import(self, endpoint, data, filename):
...
         resp = requests.post(
             endpoint,
+            timeout=self.DD_CONNECT_TIMEOUT,
             headers=self.headers,
             data=data,
             files={"file": open(filename, "rb")},  # pylint: disable=consider-using-with
         )
         if resp.status_code >= 400:
-            print(vars(resp))
+            logging.debug(vars(resp))
```
Comment (cedricbu) [inline exports/defect_dojo.py:115]: Should we also replace this `if resp.status_code >= 400:` with `try` + `resp.raise_for_status()` ? it looks much cleaner

**Review by cedricbu — COMMENTED** [review]
I'm happy to approve this commit. 
As a minor note, it would be a good thing to replace the other `if resp.status_code >= 400:` by `resp.raise_for_status()` : should we do it in this commit, or separately ?

**Comment by jeremychoi** [comment]
> should we do it in this commit, or separately ?

how about doing so in a separate commit and adding test cases for the specific functions as necessary?

**Pull request merged**